### PR TITLE
inital commit- week 2 assessment 

### PR DESCRIPTION
**Pull request by smkattoula**

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -3,15 +3,26 @@
...
 var num3 = -7
 // Expected output: "-7 is not divisble by three"
 
+const evenDivide = (num) => {
+    if (num % 3 === 0) {
+        return `${num} is divisible by three`
+    }       else {
+        return `${num} is not divisible by three`
+    }
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:22]: yay!

File: code-challenges.js [hunk 2]
```diff
@@ -21,7 +32,10 @@ var num3 = -7
 var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
 
-
+const wordsCapitalized = (array) => {
+   return array.map(value => value.charAt(0).toUpperCase() + value.substr(1));
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:37]: great!

File: code-challenges.js [hunk 3]
```diff
@@ -30,35 +44,65 @@ var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "tem
 var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
 // Expected output: [-8, 0, 8, 46, 59, 90, 107]
 
+const onlyNumbers = (array) => {
+    let newArr = []
+    newArr = array.sort((a,b) => a - b).filter(value => typeof value === "number")
+     return newArr 
 
-
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:52]: good job! you can even do this all in one line if you'd like ;)
ex: `return array.sort((a,b) => a - b).filter(value => typeof value === "number")`

File: code-challenges.js [hunk 4]
```diff
@@ -30,35 +44,65 @@ var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "tem
...
+// create a function called vowelIndex that takes in a string and logs the index of the first vowel,
+// create an array called vowelArray that contains only vowel letters.
 
+const vowelIndex = (str) => {
+   let vowelArray = ["a", "e", "i", "o", "u"]
+    str.split("")
+        for (let i= 0; i <str.length; i++)
+            if (vowelArray.includes(str[i])) {
+                return i
+            }
```
Comment (rachaelstanislaw) [inline code-challenges.js:73]: so i think you're missing one curly brace here. with some small changes to your indentation you'll be able to see where :) also, you don't really need to split the string. strings can be iterated through with loops just like arrays. cool huh?

File: code-challenges.js [hunk 5]
```diff
@@ -30,35 +44,65 @@ var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "tem
...
+    }   else if (str === "-") {
+        return num1 - num2
+    }   else if (str === "*") {
+        return num1 * num2
+    }   else if (str === "/") {
+        return num1 / num2
+    }   else if (str === "/" && num2 === 0) {
+        return "Can't divide by 0!"
+    }
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:95]: love this. think about maybe naming your operator something other than "str" perhaps? just for more clarity.

File: interview-questions.md [hunk 6]
```diff
@@ -4,57 +4,57 @@ Answer the following questions. First, without external resources. Challenge you
...
 
 
 
 3. When creating a basic (stateless) class component in React, what are the necessary elements needed to render "Hello World" on the page?
 
-  Your answer:
+  Your answer: constructor and super
 
-  Researched answer:
+  Researched answer: I'm not sure if I've understood it completely but I believe the basic elements needed are <h1></h1> ??
```
Comment (rachaelstanislaw) [inline interview-questions.md:25]: this question was a little confusing and your answer is great. but just so you know, we were looking for: import, class name extends, render, return, div/React.Fragment would be a bonus, export included in this answer somehow

File: interview-questions.md [hunk 7]
```diff
@@ -4,57 +4,57 @@ Answer the following questions. First, without external resources. Challenge you
...
 
-  Researched answer:
+  Researched answer: "JSX stands for JavaScript XML. JSX allows us to write HTML in React. JSX makes it easier to write and add HTML in React."
 
 
 
 5. What is the difference between React state and props?
 
-  Your answer:
+  Your answer: State is like the brain of a function and is dynamic. Props are the limbs and connect to the state. (still trying to graps the concept of states and props)
```
Comment (rachaelstanislaw) [inline interview-questions.md:39]: totally normal to not have a solid grasp just yet! this answer is great

File: react-challenge.html [hunk 8]
```diff
@@ -46,25 +47,32 @@
               <MoodMeter
                 moodOptions={ this.state.moods[this.state.currentMood] }
                 next={ this.nextMood }
-                setThisMood={ this.setThisMood }
-                set={ this.state.setMood }
+                set.Moods={ this.setMood }
+                this.setState={ this.setMood }
```
Comment (rachaelstanislaw) [inline react-challenge.html:51]: close!

**Review by rachaelstanislaw — COMMENTED** [review]
great job! i see a great improvement over last week as far as creating dynamic, reusable functions!

**Review by rachaelstanislaw — APPROVED** [review]
let me know if you have questions then go ahead and merge!

**Comment by smkattoula** [comment]
Thank you!!! @rachaelstanislaw

**Comment by rachaelstanislaw** [comment]
:)

**Pull request merged**